components/ws-daemon/cmd/run.go

```diff
@@ -116,3 +145,42 @@ var runCmd = &cobra.Command{
 func init() {
 	rootCmd.AddCommand(runCmd)
 }
+
+func grpcProbe(tlsConfig config.AddrTLS) func() error {
+	return func() error {
+		secopt := grpc.WithInsecure()
+		if tlsConfig.TLS != nil && tlsConfig.TLS.Certificate != "" {
+			tlsConfig, err := common_grpc.ClientAuthTLSConfig(
+				tlsConfig.TLS.Authority, tlsConfig.TLS.Certificate, tlsConfig.TLS.PrivateKey,
+				common_grpc.WithSetRootCAs(true),
+				common_grpc.WithServerName("wsdaemon"),
+			)
+			if err != nil {
+				return xerrors.Errorf("cannot load ws-daemon certificate: %w", err)
+			}
+
+			secopt = grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		conn, err := grpc.DialContext(ctx, tlsConfig.Addr, secopt)
+		if err != nil {
+			return err
+		}
+		defer conn.Close()
```

Comment:
In the current form, if there's an error, we won't close the connection:

```suggestion
conn, err := grpc.DialContext(ctx, tlsConfig.Addr, secopt)
defer conn.Close()
if err != nil {
     return err
}
```

Reply:
Suggestion is incorrect, if there is an error, there won't be a connection to close. :smile: Thanks @aledbf !